src/agent/agent.cc

```diff
@@ -68,17 +71,33 @@ namespace sp {
     }
 	
    char error_file[255],output_file[255];
-     snprintf(error_file,255,"/tmp/spi/spi-error-%d", getpid());
+     snprintf(error_file,255,"./tmp/spi/spi-error-%d", getpid());
```

Comment:
Are these directories generally available, or are they specific to how you are using SPI?

Reply:
These directories should be created upon running spi, I included bash command to create directories on demand in run.sh

However, there are many places inside SPI are currently using hardcoded directory path. To make spi portable is a TODO for me.

Reply:
I would recommend making a github issue for items like this just so we know that it's not really ready for public consumption.